testing/unittests/cryptotests.cpp

```diff
@@ -294,9 +301,100 @@ class DigiSignUnitTest : public CppUnit::TestFixture
         }
         printf("Completed executing digiSign() unit tests\n");
     }
+
+#ifdef _USE_OPENSSL
+    void pkeTests()
+    {
+        try
+        {
+            Owned<CLoadedKey> publicKey = loadPublicKeyFromMemory(pubKey, nullptr);
+            Owned<CLoadedKey> privateKey = loadPrivateKeyFromMemory(privKey, nullptr);
+
+            // create random data
+            MemoryBuffer toEncryptMb;
+            fillRandomData(244, toEncryptMb); // max for RSA
+
+            MemoryBuffer pkeMb;
+            publicKeyEncrypt(pkeMb, toEncryptMb.length(), toEncryptMb.bytes(), *publicKey);
+
+            MemoryBuffer decryptedMb;
+            privateKeyDecrypt(decryptedMb, pkeMb.length(), pkeMb.bytes(), *privateKey);
+
+            ASSERT(toEncryptMb.length() == decryptedMb.length());
+            ASSERT(0 == memcmp(toEncryptMb.bytes(), decryptedMb.bytes(), toEncryptMb.length()));
+        }
```

Comment:
You should add some tests that execute a bunch of these in a tight loop and print out the timings. See digisign unit tests....

Reply:
could do, benchmarking is not normally done in unit tests.